src/main/scala/com/advancedtelematic/campaigner/db/Campaigns.scala

```diff
@@ -91,13 +91,21 @@ protected [db] class Campaigns(implicit db: Database, ec: ExecutionContext)
         CampaignStatus.values.map(s => s -> counts.getOrElse(s, 0)).toMap
       }
 
-  def countFinished(campaignId: CampaignId): Future[Long] =
-    campaignRepo.countDevices(campaignId) { status =>
+  /**
+   * Returns the number of devices that took part in the given campaigns and
+   * finished, either successfully or with a failure.
+   */
+  def countFinished(campaignIds: Set[CampaignId]): Future[Long] =
+    campaignRepo.countDevices(campaignIds) { status =>
       status === DeviceStatus.successful || status === DeviceStatus.failed
     }
 
-  def countCancelled(campaignId: CampaignId): Future[Long] =
-    campaignRepo.countDevices(campaignId) { status =>
+  /**
+   * Returns the number of devices that took part in the given campaigns, but
+   * were cancelled without being affected
```

Comment:
`without being affected` could be misleading? If it was part of the campaign it was, by definition, affected (unless I misunderstood what are the affected devices).